src/fdb5/api/helpers/ArchiveCallback.h

```diff
@@ -20,8 +20,10 @@
 
 namespace fdb5 {
 
-using ArchiveCallback = std::function<void(const Key&, const FieldLocation&)>;
+using ArchiveCallback = std::function<void(const Key& key, const void* data, size_t length, const FieldLocation&)>;
```

Comment:
Pass the data and a promise<FieldLocation>
Note in comment that the callback must return _before_ the FieldLocation will be satisfied (so promise.get() will need to be called in a thread/async/something)

Reply:
Passing a future rather than the promise (promise allows you to set the value, future allows you to get the value. Callback should only get the value).

> callback must return before the FieldLocation will be satisfied

Personally, I think it would be better if the callback did not *have* to implemented its own thread (though we do for gribjump): the important thing is that promse.set_value(location) is called before calling the callback (when using a single-threaded fdb backend), or the thread which sets the value has been started (when using remote fdb) before calling the callback.